libmamba/src/download/downloader.cpp

```diff
@@ -239,19 +239,24 @@ namespace mamba::download
         curl_debug_callback(CURL* /* handle */, curl_infotype type, char* data, size_t size, void* userptr)
         {
             auto* logger = reinterpret_cast<spdlog::logger*>(userptr);
-            auto log = Console::hide_secrets(std::string_view(data, size));
+            std::string log;
             switch (type)
             {
                 case CURLINFO_TEXT:
+                    log = Console::hide_secrets(std::string_view(data, size));
                     logger->info(fmt::format("* {}", log));
                     break;
                 case CURLINFO_HEADER_OUT:
+                    log = Console::hide_secrets(std::string_view(data, size));
                     logger->info(fmt::format("> {}", log));
                     break;
                 case CURLINFO_HEADER_IN:
+                    log = Console::hide_secrets(std::string_view(data, size));
                     logger->info(fmt::format("< {}", log));
                     break;
                 default:
+                    // WARNING Using `hide_secrets` here will give a seg fault on linux,
```

Comment:
I actually found out that the seg fault was related to the `default` case, so getting `hide_secrets` back in the other cases and adding a warning in the default one.